Mathlib/Data/Sum/LE.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2024 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.Order.Basic
+import Mathlib.Algebra.Group.Pi.Basic
+/-!
+# TODO
+
+Interaction between `Sum.elim` and `≤`
+-/
+
+namespace Sum
+
+variable {α₁ α₂ β : Type*} [LE β]
+
+lemma elim_le_elim_iff (u₁ v₁ : α₁ → β) (u₂ v₂ : α₂ → β) :
+    Sum.elim u₁ u₂ ≤ Sum.elim v₁ v₂ ↔ u₁ ≤ v₁ ∧ u₂ ≤ v₂ := by
+  constructor <;> intro hyp
+  · constructor
+    · intro i₁
+      have hi₁ := hyp (Sum.inl i₁)
+      rwa [Sum.elim_inl, Sum.elim_inl] at hi₁
+    · intro i₂
+      have hi₂ := hyp (Sum.inr i₂)
+      rwa [Sum.elim_inr, Sum.elim_inr] at hi₂
+  · intro j
+    cases j with
+    | inl j₁ =>
+      rw [Sum.elim_inl, Sum.elim_inl]
+      exact hyp.left j₁
+    | inr j₂ =>
+      rw [Sum.elim_inr, Sum.elim_inr]
+      exact hyp.right j₂
+
+lemma const_le_elim_iff (b : β) (v₁ : α₁ → β) (v₂ : α₂ → β) :
+    (Function.const _ b) ≤ Sum.elim v₁ v₂ ↔ (Function.const _ b) ≤ v₁ ∧ (Function.const _ b) ≤ v₂ :=
+  by rw [← elim_const_const, elim_le_elim_iff]
```

Comment:
```suggestion
  elim_const_const _ ▸ elim_le_elim_iff _ _ _ _
```
I suggest term mode here (and below) because I feel like the `by` on the start of the line is kinda annoying.